### PR TITLE
[CI] Kiali working with No Istiod access test

### LINKED ISSUE #5722: [CI] Kiali working with No Istiod access test
Create testing cases for Kiali working with no istiod:

- Should be in the same pipeline if possible, that is why just the backend integration tests will be modified. 

istio_api_enabled configuration setting must be set to false

Things to test: 
- Istio api enabled is false 
- No registry services
- No proxy status 
- No validations
 
Example of testing to update config values: 
https://github.com/kiali/kiali/blob/master/tests/integration/tests/remote_istiod_test.go

REF. https://github.com/kiali/kiali/pull/5642

### PR DESCRIPTION
**Pull request by josunect**
Fixes #5722 

Integration tests

### CODE REVIEW

File: tests/integration/tests/no_istiod_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	}
+}
+
+func TestNoIstiod(t *testing.T) {
+	update_istio_api_enabled(false)
+	t.Run("ServicesListNoRegistryServices", servicesListNoRegistryServices)
+	t.Run("NoProxyStatus", noProxyStatus)
+	t.Run("istioStatus", istioStatus)
+	t.Run("emptyValidations", emptyValidations)
+	update_istio_api_enabled(true)
```
Comment (hhovsepy) [inline tests/integration/tests/no_istiod_test.go:72]: I will suggest move this line as "defer update_istio_api_enabled(true)" before disabling api call.

**Comment by josunect** [comment]
> @josunect sorry for late review, I have only one change suggestion. Otherwise looks good, thank you.

No late at all! Thanks for the review, @hhovsepy ! I will make the suggested change.

**Review by leandroberetta — APPROVED** [review]
LGTM

**Comment by josunect** [comment]
> LGTM

Thanks for the review!

**Pull request merged**